orderbook/src/orderbook.rs

```diff
@@ -276,58 +308,71 @@ mod tests {
             .in_sequence(&mut seq)
             .return_const(Some(another_balance));
 
-        set_order_balance(orders.as_mut_slice(), &balance_fetcher).await;
-        assert_eq!(orders[0].order_meta_data.available_balance, Some(a_balance));
+        let balances = track_and_get_balances(&balance_fetcher, orders.as_slice()).await;
         assert_eq!(
-            orders[1].order_meta_data.available_balance,
-            Some(another_balance)
+            balances,
+            hashmap! {
+                (owner, a_sell_token) => a_balance,
+                (owner, another_sell_token) => another_balance
+            }
         );
     }
 
     #[tokio::test]
     async fn filters_insufficient_balances() {
-        let mut orders = vec![
+        let mut balance_fetcher = MockBalanceFetching::new();
+        balance_fetcher
+            .expect_get_balance()
+            .return_const(Some(10.into()));
+
+        let orders = vec![
             Order {
                 order_creation: OrderCreation {
-                    sell_amount: 100.into(),
-                    partially_fillable: true,
+                    sell_amount: 3.into(),
+                    fee_amount: 3.into(),
                     ..Default::default()
                 },
                 order_meta_data: OrderMetaData {
-                    available_balance: Some(50.into()),
+                    creation_date: DateTime::<Utc>::from_utc(
+                        NaiveDateTime::from_timestamp(2, 0),
+                        Utc,
+                    ),
                     ..Default::default()
                 },
             },
             Order {
                 order_creation: OrderCreation {
-                    sell_amount: 200.into(),
-                    partially_fillable: false,
+                    sell_amount: 1.into(),
```

Comment:
nanonit: any reason we want three orders here? I feel it might be slightly easier to understand with 2 orders.